File: Thales.Web/src/app/reports/report.ts
import { Application } from '../application/application';

export class Report {
    id: string;
    title: string;
    summary: string;
    notes: string;
    url: string;
    submitDate: Date;
    updated: Date;
    isRead: boolean;
    completed: Date;
    reportType: ReportType = 0;
    priority: Priority = 0;
    imageId: string;
    applicationId: string;
    application: Application;
    OnwerId: string;
    owner: Onwer;
    constructor() {
        this.application = new Application();
    }

}

export class Onwer {
    name: string;
    lastname: string;
}

export enum Priority {
    Χαμηλό,
    Μεσαίο,
    Υψηλό,
    Επείγον,
    Όλα
}

export enum ReportType {
    Bug,
    Feature,
    Change,
    Support,
    All
}

File: BugLogThalis/Models/Report.cs
using BugLogThalis.Models.Views;
using System;

namespace BugLogThalis.Models
{
    public class Report : Entity
    {
        public string Title { get; set; }
        public string Summary { get; set; }
        public string Notes { get; set; }
        public string Url { get; set; }
        public DateTime SubmitDate { get; set; }
        public DateTime Updated { get; set; }
        public bool IsRead { get; set; }
        public DateTime? Completed { get; set; }
        public ReportType ReportType { get; set; }
        public Priority Priority { get; set; }
        public string ImageId { get; set; }
        public string ApplicationId { get; set; }
        public string OnwerId { get; set; }
        public virtual UserView Onwer { get; set; }

    }

    public enum ReportType
    {
        Bug,
        Feature,
        Change,
        Support,
        All
    }

    public enum Priority
    {
        Low,
        Normal,
        High,
        Urgent,
        All
    }
}
